### PR TITLE
Get multiple resource per schema example running

### PR DESCRIPTION
**Pull request by patricksrobertson**
This does the following:
* Documents the project so that we can all remember wtf is going on.
* Adds `Braise.Schema` which reflects parsed JSON of the `schema.json` file.
* Adds `Braise.Schema.resource/1` which parses the schema file and generates `Braise.Resource` structs for each RESTful resource in the schema file.
* Recurses through each resource and generates adapters/models for each resource.
* Eliminates the `Braise.DefinitionParser` as I just made that a part of the `Braise.Resource` struct.
* Renames the `Braise.ResponseParser` to `Braise.ResourceDefinitionTuples` as it is a struct that contains tuples of response attributes to definition attributes.

### CODE REVIEW

File: lib/braise/cli.ex [hunk 1]
```diff
@@ -37,18 +37,22 @@ defmodule Braise.CLI do
...
     {:ok, file} = File.read(filename)
 
-    resource       = Poison.decode!(file, as: Braise.Resource)
-    {:ok, name, adapter} = Braise.AdapterTemplate.generate_from_resource(resource)
+    schema = Poison.decode!(file, as: Braise.Schema)
+    resources = Braise.Schema.resources(Map.to_list(schema.properties), schema.links, schema.definitions)
 
-    adapter_filename = output <> "/addon/adapters/" <> name <> ".js"
-    write_to_file(adapter, adapter_filename)
+    for resource <- resources do
```
Comment (patricksrobertson) [inline lib/braise/cli.ex]: should recurse this.

File: lib/braise/definition_parser.ex [hunk 2]
```diff
@@ -1,7 +1,6 @@
 defmodule Braise.DefinitionParser do
```
Comment (patricksrobertson) [inline lib/braise/definition_parser.ex]: this module serves no purpose.

File: lib/braise/schema.ex [hunk 3]
```diff
@@ -0,0 +1,21 @@
+defmodule Braise.Schema do
+  defstruct [:definitions, :description, :id, :links, :properties, :title, :type]
+
+  def keys(schema) do
```
Comment (patricksrobertson) [inline lib/braise/schema.ex]: think this was me plane debugging.

File: lib/braise/cli.ex [hunk 4]
```diff
@@ -49,6 +55,8 @@ defmodule Braise.CLI do
     |> Braise.ModelToEmberModel.convert
     |> Braise.EmberModelTemplate.generate
     |> write_to_file(model_filename)
+
+    write_adapters_and_models(tail, output)
```
Comment (alexrothenberg) [inline lib/braise/cli.ex]: love the tail recursion

**Comment by alexrothenberg** [comment]
i think it looks good but don't feel i'm yet qualified to know

Only question is on the org of the examples dir.  Do you think separating into 2 subdirs `input/output` or `source/generated` or something like that would make clearer what is being fed into braise and what its producing?

:octocat:

**Comment by patricksrobertson** [comment]
Can do on the examples.

**Pull request merged**